app/api/views/brief_responses.py

```diff
@@ -0,0 +1,116 @@
+from flask import abort, jsonify
+from flask_login import current_user, login_required
+from app.api import api
+from ...models import AuditEvent
+from app.api.services import brief_responses_service, audit_service
+from ...datetime_utils import utcnow
+from dmapiclient.audit import AuditTypes
+import rollbar
+
+
+@api.route('/brief-response/<int:brief_response_id>/withdraw', methods=['POST'])
+@login_required
+def withdraw_brief_response(brief_response_id):
+    """Withdraw brief responses (role=supplier)
+    ---
+    tags:
+      - "Brief Responses"
+    security:
+      - basicAuth: []
+    parameters:
+      - name: brief_response_id
+        in: path
+        type: number
+        required: true
+    definitions:
+      BriefResponse:
+        type: object
+        properties:
+          id:
+            type: number
+          data:
+            type: object
+          brief_id:
+            type: number
+          supplier_code:
+            type: number
+    responses:
+      200:
+        description: Successfully withdrawn a candidate
+      400:
+        description: brief_response_id not found
+    """
+    brief_response = (brief_responses_service
+                      .find(id=brief_response_id,
+                            supplier_code=current_user.supplier_code,
+                            withdrawn_at=None)
+                      .one_or_none())
+
+    withdrawn_at = utcnow()
+    if brief_response:
+        brief_response.withdrawn_at = withdrawn_at
+        brief_responses_service.save(brief_response)
+    else:
+        abort(404)
```

Comment:
not a massive fan of just returning `404` - what if there was a response with id, but it was already withdrawn, or didnt belong to logged in supplier?  You may want to return a useful message (even if not used on FE)